crypto/evp/e_chacha20_poly1305.c

```diff
@@ -328,8 +330,9 @@ static int chacha20_poly1305_ctrl(EVP_CIPHER_CTX *ctx, int type, int arg,
     switch(type) {
     case EVP_CTRL_INIT:
         if (actx == NULL)
-            actx = ctx->cipher_data
-                 = OPENSSL_zalloc(sizeof(*actx) + Poly1305_ctx_size());
+            EVP_CIPHER_CTX_set_cipher_data(ctx,
+                OPENSSL_zalloc(POLY1305_actx_size(actx)));
+            actx = EVP_CIPHER_CTX_get_cipher_data(ctx);
```

Comment:
May I suggest this instead?

``` C
            actx = OPENSSL_zalloc(sizeof(*actx) + Poly1305_ctx_size());
            EVP_CIPHER_CTX_set_cipher_data(ctx, actx);
```

Looks a bit cleaner, doesn't it?

Reply:
Thank you for the great solution 👍